lib/Dialects/LinalgTransform/Transforms/TransformInterpreter.cpp

```diff
@@ -546,26 +668,73 @@ findTransformTarget(OpTy transformOp, ModuleOp module,
   return applyPatternsOnce(module, std::move(patterns));
 }
 
+template <typename OpTy>
+static std::enable_if_t<!llvm::is_detected<has_target_matcher_t, OpTy>::value,
+                FailureOr<SmallVector<Operation *>>>
+matchUsingPDL(OpTy transformOp, ModuleOp module) {
+  assert(false && "matching with PDL on an unsupported op");
+}
+
+/// Returns the operations that `transformOp` should be run on. The
+/// `transformOp` is expected to have the TargetableTransformOpTrait. Its target
+/// may be either an SSA value that corresponds to a list of ops produced by
+/// other transforms and stored in `operations` or a list of ops that match the
+/// PDL pattern specified by its name.
+template <typename OpTy>
+static FailureOr<SmallVector<Operation *>>
+findTransformTarget(OpTy transformOp, ModuleOp module,
```

Comment:
should this be using an interface and drop the templating?

Reply:
This can be detemplatized with #218 AFAICS.

Reply:
great!